tree.go

```diff
@@ -1772,17 +1783,55 @@ func (n *LeafNode) serializeLeafWithUncompressedCommitments(cBytes, c1Bytes, c2B
 				children = append(children, padding...)
 			}
 		}
+
+		if isEoA {
```

Comment:
I see, so in theory this should be named `maybeIsEoA` since someone could craft some storage slots to match this pattern. But should be fine.

(this is a jk comment, `isEoA` is fine)